### DISCUSSION TITLE
Add/manage custom headers in JWT Token

### DISCUSSION THREAD

**Discussion by customsoft** [phalcon/cphalcon#16388] [Q&A]
In Phalcon ^5.2 how can you add / manage / customize the headers of a JWT Token?
I have to add the value "kid" in the header; that contains the id of public key to validate the request.

**Comment by sinbadxiii** [comment]
Hi @customsoft 
You may see example app https://github.com/sinbadxiii/phalcon-auth-jwt-example with https://github.com/sinbadxiii/phalcon-auth-jwt

**Reply by customsoft** [reply]
From a quick analysis of the code, the only point where I see data being added is here:
https://github.com/sinbadxiii/phalcon-auth-jwt-example/blob/44990cf135d70b97b96ac346f2cb57a38830ac1b/public/index.php#L77
But this data is in the payload, not in the JWT header.
Should I look elsewhere?

**Reply by sinbadxiii** [reply]
@customsoft Why do you want to change header? It header typically consists only from two parts: the type of the token, which is JWT and the signing algorithm being used,

**Reply by customsoft** [reply]
It is required to access the APIs of the Italians Public Administrations via PDND
https://docs.pagopa.it/interoperabilita-1/manuale-operativo/utilizzare-i-voucher

**Comment by niden** [comment]
@customsoft For now this is not available. I will add the functionality to allow for extending the Builder and adding headers

**Reply by customsoft** [reply]
I really appreciate it :-)
